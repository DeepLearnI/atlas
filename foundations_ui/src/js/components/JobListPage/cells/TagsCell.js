import React, { Component } from 'react';
import PropTypes from 'prop-types';

class TagsCell extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isError: this.props.isError,
      rowNumber: this.props.rowNumber,
      tags: this.props.tag,
    };
  }

  render() {
    const { tags, isError, rowNumber } = this.state;

    const pClass = isError
      ? `job-cell tag-cell error row-${rowNumber}`
      : `job-cell tag-cell row-${rowNumber}`;

    const tagList = [];
    const someTagKeys = Object.keys(tags);

    someTagKeys.map((keyName) => {
      // console.log('tags.key', tags[keyName]);
      tagList.push(<div className="tagBlock">{keyName}:{tags[keyName]}</div>);
    });

    return (
      <div className={pClass}>
        <div className="tagWrapper">
          {tagList}
        </div>
      </div>
    );
  }
}

TagsCell.propTypes = {
  tag: PropTypes.object,
  isError: PropTypes.bool,
  rowNumber: PropTypes.number,
};

TagsCell.defaultProps = {
  tag: {},
  isError: false,
  rowNumber: 0,
};

export default TagsCell;
