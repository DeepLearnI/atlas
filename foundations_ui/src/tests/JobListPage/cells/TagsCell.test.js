import React from 'react';
import ReactDOM from 'react-dom';
import TagsCell from '../../../js/components/JobListPage/cells/TagsCell';
import { shallow } from 'enzyme';
import configureTests from '../../setupTests';

configureTests();

it('Shallow Renders Tags Cell', () => {
  const wrapper = shallow(<TagsCell/>);
  expect(wrapper).toMatchSnapshot();
});