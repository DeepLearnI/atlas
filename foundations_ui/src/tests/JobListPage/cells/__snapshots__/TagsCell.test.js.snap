// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Renders Tags Cell 1`] = `
<div
  className="job-cell tag-cell row-0"
>
  <div
    className="tagWrapper"
  />
</div>
`;
